### PR TITLE
replace circleci to use github actions and add internal job for deployment

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
some updates:
- replace circleci to use github actions
- add an internal pipeline to deploy

notes:
- the circleci for this repository is already disabled
- circleci context removed
- gitlab mirror in place
- configuration for gitlab in place

#### Ticket Link
n/a

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.7'
+
+    - name: Run Python deps
+      run: |
+        pip install pipenv
```
Comment (spirosoik) [inline .github/workflows/build.yml:27]: could we add this piece in makefile? `make python-deps` (something similar)

**Review by spirosoik — APPROVED** [review]
one minor comment. other than that all good

**Review by cwarnermm — APPROVED** [review]
Thanks!

File: .github/workflows/build.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/build.yml:27]: sure thing

**Review by jwilander — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/build.yml]: done

**Review by spirosoik — APPROVED** [review]

**Pull request merged**